plugins/common/lambda-rules-registration.cpp

```diff
@@ -0,0 +1,70 @@
+#include "lambda-rules-registration.hpp"
+
+#include <iostream>
+#include <map>
+#include <memory>
+#include <string>
+
+#include "wayfire/core.hpp"
+#include "wayfire/object.hpp"
+#include "wayfire/nonstd/observer_ptr.h"
+#include "wayfire/parser/lambda_rule_parser.hpp"
+#include "wayfire/rule/lambda_rule.hpp"
+
+namespace wf
+{
+
+nonstd::observer_ptr<lambda_rules_registrations_t> lambda_rules_registrations_t::getInstance()
+{
+    auto instance = get_core().get_data<lambda_rules_registrations_t>();
```

Comment:
A much easier way to achieve this is to use `get_data_safe` (which does exactly what you are doing here).

Reply:
I looked into this, and I found that get_data_safe needs the constructor of the object to be public. This violates the intent of the registrations a bit since I wanted it to be a singleton.

I'll leave this be for now unless you tell me to change it.